api_types.ts

```diff
@@ -546,6 +546,20 @@ export interface CommonPackFormulaDef<T extends ParamDefs> {
    * @hidden
    */
   readonly allowedAuthenticationNames?: string[];
+
+  /**
+   * The JavaScript function that implements parameter validation
+   *
+   * This function takes in parameters and a context containing permissions mode information,
+   * and validates the parameters.
+   *
+   * @returns if the parameters are valid
+   * throws {@link ParameterValidationError} if the parameters are invalid.
+   */
+  validateParameters?(
+    params: ParamValues<ParamDefs>,
+    context: ExecutionContext | SyncExecutionContext,
```

Comment:
Could maybe follow up with this because the parameterization is annoying, but I think Oleg has successfully made our usage of the *Context types correct via generics and it would be nice not to regress that. That is, this isn't actually a union type, it's exactly one or the other, depending on whether this is a sync or not. So in other places you have usage like `MetadataFormula<ContextT extends ExecutionContext = ExecutionContext>`